### DISCUSSION TITLE
Purchase agreement instead of software

### DISCUSSION THREAD

**Discussion by Deedikjupijn** [immich-app/immich#11384] [Feature Request]
Something I'd love to see would be a way for users to buy into a legally binding agreement with futo that forces them to keep the current license (AGPL 3) or just keeping future updates open source with included that they will not paywall any features for a couple of years after the purchase (this duration would be in the contract, not sure how long or how exactly it could be worded). 

This would allow for a continuous stream of income without removing any features from those who aren't able to pay for whatever reason.

Curious what others think and hope to see this implemented. As always, thanks for reading and hope you have a nice day!

**Comment by mertalev** [comment]
This was something we discussed, but we decided against it. While there are merits to this, the concept is confusing even for us and would inevitably raise more questions than it answers.

Another alternative we explored is to instead design a contract that forbids us from adding paywalls, but the initial legal feedback we got is that this doesn't really work. A sweeping, perpetual promise like "no paywalls" just isn't legally enforceable.

We empathize with the concerns here and know they come from past experiences. We hope our steps to move away from the *The-Words-Which-Must-Not-Be-Named* in response to community feedback can alleviate these concerns regardless.

**Reply by Deedikjupijn** [reply]
Hey great that it already has been discussed.

I would like to know how it raises more questions then answers. I think it would set the intentions very clearly while also making the purchase worth something. Minecraft used to have in the license agreement that with a single purchase you would get any and all future versions of Minecraft (not sure but I think microsoft changed this).

At the current moment it does not really feel like I'm buying anything. as it's hard to see where the money is going development wise, yes a lot of updates but still it's not the thing the average user sees. You could argue a working product but the product doesn't change from working to broken if one person doesn't pay (I get its a lot of small payments that make up for the couple of people that do not pay and this makes the product work). 

What my goal was with this discussion is that there would be value added to the purchase without locking any features away from the users that do not pay.

Curious to see if there is any other ideas that might improve on this to create gives value to the purchase.

As always, thanks for reading and hope you have a great day! 👋

**Reply by mertalev** [reply]
I suppose it depends on the wording, but the premise of selling a guarantee of AGPL can give the impression that the user has to buy it to get Immich under AGPL, or that we're trying to trick them into thinking this, or that we intend to move away from AGPL and make it a paid privilege. The average user likely wouldn't understand the benefit of it either - it's a niche topic that feels odd in a sales pitch.

In general, you can think of the purchase phrasing as being in exchange for continued development. It's weird, yes - there's no immediate value proposition to it - but it's an experiment we're trying out.

I'll also just say that while we're happy to have made improvements in this area, there isn't much appetite in revisiting the payment discussion 😅, at least right now. It took a lot of discussion and work to get this far, and we're more excited to get back to feature development.
